### PR TITLE
Fix: handle upload stream with unknown size

### PR DESCRIPTION
**Pull request by Hydrog3n**
Hello, first thank you for the lib 🙏🏼 

I arrived on a case where I have a stream from a DB who are converted to csv through pipe and uploading to S3.  

But in certain case I have no data and I got this error since I update from 8.0.5 => 8.0.7 

```
S3Error: The XML you provided was not well-formed or did not validate against our published schema
```

I'm open to improvement. 

## Bug Fixes
* Improved multipart upload handling with refined part number sequencing and matching logic for better reliability during large file uploads.
* Enhanced fallback mechanism for edge cases that may occur during the upload process.

### CODE REVIEW

**Comment by prakashsvmx** [comment]
Please add unit/functional test @Hydrog3n

**Comment by Hydrog3n** [comment]
@prakashsvmx Yep I was on it sorry 🙂

File: src/internal/client.ts [hunk 1]
```diff
@@ -1800,22 +1800,20 @@ export class TypedClient {
         chunkier.on('end', resolve).on('error', reject)
       }),
       (async () => {
-        let partNumber = 1
+        let partNumber = 0
 
         for await (const chunk of chunkier) {
           const md5 = crypto.createHash('md5').update(chunk).digest()
 
+          partNumber++
```
Comment (prakashsvmx) [inline src/internal/client.ts:1808]: This needs to be compatibility tested for partNumber

**Review by prakashsvmx — CHANGES_REQUESTED** [review]
Please test with minio, aws s3 on both and present results

**Review by prakashsvmx — APPROVED** [review]
